src/Raven.Server/Documents/TransactionOperationsMerger.cs

```diff
@@ -848,5 +1125,283 @@ public void Dispose()
+                    return jsonSerializer.Deserialize<UpdateSiblingCurrentEtagDto>(reader);
+                case nameof(IncomingReplicationHandler.MergedUpdateDatabaseChangeVectorCommand):
+                    return jsonSerializer.Deserialize<MergedUpdateDatabaseChangeVectorCommandDto>(reader);
+                case nameof(AdminRevisionsHandler.DeleteRevisionsCommand):
+                    return jsonSerializer.Deserialize<DeleteRevisionsCommandDto>(reader);
+                case nameof(RevisionsOperations.DeleteRevisionsBeforeCommand):
+                    throw new ReplayTransactionsException(
+                        "Because this command is deleting according to revisions' date & the revisions that created by replaying have different date an in place decision needed to be made",
+                        peepingTomStream);
+                case nameof(TombstoneCleaner.DeleteTombstonesCommand):
+                    return jsonSerializer.Deserialize<DeleteTombstonesCommandDto>(reader);
+                case nameof(OutputReduceIndexWriteOperation.OutputReduceToCollectionCommand):
+                    return jsonSerializer.Deserialize<OutputReduceToCollectionCommandDto>(reader);
+                default:
+                    throw new ReplayTransactionsException($"Can't read {type} for replay", peepingTomStream);
+            }
+        }
+
+        private static JsonSerializer GetJsonSerializer()
+        {
+            var jsonSerializer = DocumentConventions.Default.CreateSerializer();
+            jsonSerializer.Converters.Add(SliceJsonConverter.Instance);
+            return jsonSerializer;
+        }
+
+        public class ReplayTransactionsException : Exception
+        {
+            public string Context { get; }
+
+            public ReplayTransactionsException(string message, PeepingTomStream peepingTomStream)
+            : base(message)
+            {
+                try
+                {
+                    Context = Encodings.Utf8.GetString(peepingTomStream.PeepInReadStream());
+                }
+                catch (Exception e)
+                {
+                    Context = "Failed to generated peepedWindow: " + e;
+                }
+            }
+
+            public override string Message => base.Message + ";\rContext:\r" + Context;
+        }
+
+        private class RecordingDetails
+        {
+            public string Type { get; }
+            public DateTime DateTime { get; }
+
+            public RecordingDetails(string type)
+            {
+                Type = type;
+                DateTime = DateTime.Now;
+            }
+        }
+
+        private class StartRecordingDetails : RecordingDetails
+        {
+            private const string DetailsType = "StartRecording";
+            public string Version { get; }
+
+            public StartRecordingDetails()
+            : base(DetailsType)
+            {
+                Version = ServerVersion.FullVersion;
```

Comment:
We typically use RavenVersionAttribute.Instance.Build only

Reply:
done